File: src/main/tsx/types/CategoryDTO.ts
import {CategorySmoothType} from "@/types/CategorySmoothType";
import {TransactionType} from "@/types/TransactionType";

export interface CategoryDTO {
    id: number;
    parentId: number;
    label: string;
    transactionType: TransactionType;
    smoothType: CategorySmoothType;
    start: string;
    end: string;
    target: number;
}

File: backend/src/main/java/com/lennartmoeller/finance/dto/CategoryDTO.java
package com.lennartmoeller.finance.dto;

import com.lennartmoeller.finance.model.CategorySmoothType;
import com.lennartmoeller.finance.model.TransactionType;
import java.util.List;
import javax.annotation.Nullable;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@RequiredArgsConstructor
@Setter
public class CategoryDTO {
    private @Nullable Long id;
    private @Nullable Long parentId;
    private String label;
    private TransactionType transactionType;
    private CategorySmoothType smoothType;
    private @Nullable String icon;
    private List<TargetDTO> targets;
}
